File: 3.EDD/5°Version/Cola/C++/Financiera.cpp
#include <iostream>
#include <queue>
#include <cstdlib>
#include "Prestamo.h"
using namespace std;

int main(){
    system("color 05");
    queue<Prestamo> coleccion;
    int continuar;

    string palabras;
    double cantidad;
    Prestamo P1;

    do{
        // Datos del acreedor-------------------
        cout << "Ingresar datos del acreedor";
        cout << "\n \n";
        cout << "Ingresar nombres: ";
        getline(cin,palabras);
        P1.setNombre(palabras);
        cout << "Ingresar apellidos: ";
        getline(cin,palabras);
        P1.setApellido(palabras);
        cout << "Ingresar un numero telefonico: ";
        getline(cin,palabras);
        P1.setTelefono(palabras);
        cout << "Ingresar correo electronico (Opcional): ";
        getline(cin,palabras);
        P1.setCorreo(palabras);

        //Datos del pretamo-------------------------
        cout <<"\n \n";
        cout << "Ingresar la cantidad a prestar: ";
        cin >> cantidad;
        P1.setPrestamo(cantidad);
        cout << "Ingresar el porcentaje de interes (%): ";
        cin >> cantidad;
        P1.setPorcentaje(cantidad);
        cout << "Ingresar la cantidad de semanas a pagar: ";
        cin >> cantidad;
        P1.setSemana(cantidad);

        coleccion.push(P1);
        cout << "Prestamos registrado...... \n\n";
       
        cout << "Presiona 1 Para continuar y o para finalizar: ";
        cin >> continuar;
        cin.ignore();

    }while(continuar == 1);

    cout << "\nLista de prestamos capturados del dia:\n";
    cout << "\n-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=\n";
    while(!coleccion.empty()){
        Prestamo P1= coleccion.front();
        cout << P1.toString();
        coleccion.pop();
    }
   
    return 0;
}

File: 3.EDD/5°Version/Cola/Java/Financiera.java
import java.util.Scanner;
import java.util.LinkedList;;
public class Financiera {
    public static void main (String[] args){
        Scanner mex = new Scanner (System.in);
        LinkedList<Prestamo> cola= new LinkedList<>();
        int continuar =1;

        while(continuar != 0){
            Prestamo P1 = new Prestamo();

            System.out.println("\n \n");
            System.out.println("Datos del acreedor");
            System.out.println("\n");
            System.out.println("Ingresa los nombres: ");
            P1.setNombre(mex.nextLine());
            System.out.println("Ingresa los apellidos: ");
            P1.setApellido(mex.nextLine());
            System.out.println("Ingresa un numero telefonico: ");
            P1.setTelefono(mex.nextLine());
            System.out.println("Ingresa un correo electronico (Opcional): ");
            P1.setCorreo(mex.nextLine());

            System.out.println("Ingrese la cantidad a prestar: ");
            P1.setPrestamo(mex.nextDouble());
            System.out.println("Ingresar el porcentaje de interes: ");
            P1.setPorcentaje(mex.nextDouble());
            System.out.println("Ingresa la cantidad de a pagar: ");
            P1.setSemana(mex.nextDouble());

            cola.addLast(P1);

            System.out.println("El prestamo a sido registrado");

           
            System.out.println("Para continuar oprima 1, para finalizar 0: ");
            continuar = mex.nextInt();
            mex.nextLine();
        }

        System.out.println("Extrayendo los elementos de la cola.\n");
        while (!cola.isEmpty()) {
            Prestamo iterador = cola.removeFirst();
            System.out.println(iterador.ToString());
        }
       
        mex.close();
    }
}
